addons/misra.py

```diff
@@ -133,15 +139,15 @@ def bitsOfEssentialType(expr):
     if type is None:
         return 0
     if type == 'char':
-        return CHAR_BIT
+        return typeBits['CHAR']
```

Comment:
I only made the `typeBits` dictionary because all the `*_BIT` variables are global but weren't explicitly defined as such. The variables aren't changed but they are set once by a function. The `'CHAR'` is a key for the dictionary (kind of like a struct) so it's not much different than several global variable names. Would it be clearer to explicitly initialize it as 

    typeBits = {
        'CHAR': None,
        'SHORT': None,
        'INT': None,
        'LONG': None,
        'LONG_LONG': None
    }

Reply:
I hear you! Type safety is certainly a problem with Python. Javascript has the same problem but Typescript was invented to try and solve it. In Python, the conventional wisdom is to explicitly describe your data structures in a docstring for each function and avoid global variables. Not sure what else you can do really. This addon may have been better suited as a class and these variables would be set on initialization, but this problem still exists.